Automattic/wp-calypso
# Adds new Blaze origin for atomic sites using classic view

With the work done in the menu unification, we now have the possibility of Atomic sites using the Jetpack Blaze version of the dashboard. We want to distinguish those cases from regular self-hosted jetpack-connected sites.

## Proposed Changes
* Added the new Blaze origin `jetpack-atomic` (we use the site option `is_wpcom_atomic` to differentiate atomic sites)
* Fixed some events that triggered a `calypso_dsp_widget_start` (Blaze create campaign modal) when they redirected the user to the Blaze landing page.
* Fixed the entry point in the Stats module for self-hosted sites. I found this error while working on this change, if you click the Promote button shown in the stats module page `stats/day/posts/{SITE}`, you are not getting redirected to the Jetpack Blaze dashboard.

## Testing Instructions

* Open the browser network console, and filter by all request with name `t.gif`
* Navigate to your site's home page (`home/{SITE}`)
* Click on the `Promote with Blaze` quick link, and verify that we trigger a event with the name `calypso_customer_home_my_site_quick_link_blaze`
* Go to the posts page, and click on the Promote with Blaze option for any of your posts
* Verify that the event `calypso_dsp_widget_start ` is triggered with the entrypoint `posts-meatball-menu`
* Click on the Stats, scroll down and click on View details of the Posts & Pages section
* Find a time period were you have some stats, and hover over any of the posts and click the Promote button.
* Verify that the widget is loaded and that the event `calypso_dsp_widget_start ` is triggered with the entrypoint `mysites_stats_posts-and-pages_speaker-button`

**Test the new origin, and specific changes of Jetpack Stats fix in entry point**

You will need a local jetpack environment working.

* Check out this branch
* In a terminal navigate to `wp-calypso/apps/blaze-dashboard`
* Execute `BLAZE_DASHBOARD_PACKAGE_PATH=/PATH_TO_JETPACK_REPO/projects/packages/blaze yarn dev`
You need to change `PATH_TO_JETPACK_REPO` with the path of the jetpack local repository
* Open another terminal and navigate to `wp-calypso/apps/odyssey-stats`
* Execute `STATS_PACKAGE_PATH=/PATH_TO_JETPACK_REPO/projects/packages/stats-admin yarn dev`
You need to change `PATH_TO_JETPACK_REPO` with the path of the jetpack local repository
* In another console, go to the jetpack repository and execute `jetpack build` and then choose plugins -> jetpack. Choose to compile with all of its dependencies. 
* Open a tunnel pointing to your local jetpack environment and access the wp-admin page using the public name of the tunel
* Go to Jetpack->Stats, scroll down and click on View details of the Posts & Pages section
* Find a time period were you have some stats, and hover over any of the posts and click the Promote button.
* Verify that you are correctly redirected to the Jetpack Blaze page -> Create campaign widget for the post you choose.
* In your editor, open [this file](https://github.com/Automattic/jetpack/blob/97931d43cc1d9b8b59f6bd832b250e9b4e3f3a7c/projects/packages/blaze/src/class-dashboard-config-data.php#L76) and add this additional line (this will set your site as atomic):
```
'is_wpcom_atomic' => true,
```
* In your browser, go to Tools->Advertising and verify that thevents we are triggering have the `jetpack-atomic` origin.


## Pre-merge Checklist
